src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -427,18 +427,17 @@ free_unmap_bo(xclDeviceHandle handle, xclBufferHandle boh, void * boptr, size_t
  * helper function for P2P test
  */
 static bool
-p2ptest_set_or_cmp(char *boptr, size_t size, char pattern, bool set)
+p2ptest_set_or_cmp(char *boptr, size_t size, const std::vector<char>& valid_data, bool set)
 {
   int stride = xrt_core::getpagesize();
 
   assert((size % stride) == 0);
+  assert(size > valid_data.size());
   for (size_t i = 0; i < size; i += stride) {
-    if (set) {
-      boptr[i] = pattern;
-    }
-    else if (boptr[i] != pattern) {
+    if (set)
+      std::memcpy(&(boptr[i]), valid_data.data(), valid_data.size());
```

Comment:
This function is writing character(s) at each page boundary. is it expected? You just want to verify only only some characters at page_boundary to reduce runtime of this test? If yes, better to add one comment explaining this

if valid_data size is more than page size, then you are overwriting boptr. ( Thats not the case here in Linux as you are sending maximum 1k data size whereas page size is 4k. Better to add explicit comment)

Reply:
The purpose of this function is to write/check some piece of data in all pages within the given size constraint. The size given by the user determines how many pages will be tested. It would be better to pass the number of pages to test.

I will put a check for that, I could see an issue with writing over the boundary